arangod/Replication2/Methods.cpp

```diff
@@ -748,35 +782,42 @@ struct ReplicatedLogMethodsCoordinator final
       }
       return network::sendRequest(pool, "server:" + participant,
                                   fuerte::RestVerb::Post, path, buffer, opts)
-          .thenValue([participant](network::Response&& resp) {
-            return resp.combinedResult().withError([&](auto& err) {
-              return err.appendErrorMessage("; compaction for participant " +
-                                            participant);
-            });
+          .thenValue([participant](
+                         network::Response&& resp) noexcept -> ResultPair {
+            auto result = resp.deserialize<CompactionResultMap>();
+            if (result.fail()) {
+              return std::make_pair(
+                  participant, CompactionResponse::fromResult(result.result()));
+            }
+            TRI_ASSERT(result->size() == 1 && result->contains(participant));
+            return std::make_pair(participant, result->at(participant));
           });
     };
 
-    std::vector<futures::Future<Result>> futs;
+    std::vector<futures::Future<ResultPair>> futs;
     for (auto const& [participant, p] : spec->participantsConfig.participants) {
       futs.emplace_back(compactParticipant(participant));
     }
 
-    return futures::collectAll(std::move(futs)).thenValue([](auto&& results) {
-      // return first error
-      for (auto const& tryRes : results) {
-        if (auto res = tryRes.get(); res.fail()) {
-          return res;
-        }
-      }
-      return Result{};
-    });
+    return futures::collectAll(std::move(futs))
+        .thenValue([](std::vector<futures::Try<ResultPair>>&& results)
+                       -> CompactionResultMap {
+          CompactionResultMap map;
```

Comment:
We're currently not doing micro optimizations while the code is still changing as much as it does.

Reply:
It's not micro optimization it's just correct way to write code:
if you make cycle and push to container is common to make reserve before it.

Even in clang-tidy exist such rule, so it's same as speak it's ok to use c-style cast here instead of static_cast

https://clang.llvm.org/extra/clang-tidy/checks/performance/inefficient-vector-operation.html

Reply:
And fix it probably easier than writing answer to me :)

Reply:
True. But you can turn that argument around as well. ;)

And of course you're right that a reserve is sensible from a performance perspective, I don't argue against that at all. But again, this is a debate of principles from my POV. The performance of these lines is just so way down on the list of priorities here that I don't bother. This code sends a network request to each participant, each of which possibly runs a log compaction, then waits for the answers. On the other hand, we're arguing about one or two additional (tiny) allocations. So I estimate we're talking about at least three, probably four, orders of magnitude difference.

Reply:
Yep